src/components/Table/hoc/withTableSettings/TableColumnSetup/TableColumnSetup.tsx

```diff
@@ -1,270 +1,154 @@
-
-type Item = TableColumnSetupItem;
+const tableColumnSetupCn = b(null);
+const applyButtonCn = b('apply');
+const requiredDndItemCn = b('required-item');
 
 interface SwitcherProps {
     onKeyDown: React.KeyboardEventHandler<HTMLElement>;
     onClick: React.MouseEventHandler<HTMLElement>;
 }
 
+type Item = TableColumnSetupItem & DndTreeSelectItem;
+
 export interface TableColumnSetupProps {
-    // for Button
-    disabled?: boolean;
-    /**
-     * @deprecated Use renderSwitcher instead
-     */
-    switcher?: React.ReactElement | undefined;
     renderSwitcher?: (props: SwitcherProps) => React.ReactElement | undefined;
 
-    // for List
-    items: Item[];
-    itemsHeight?: number | ((items: Item[]) => number);
+    items: TableColumnSetupItem[];
     sortable?: boolean;
-    filterable?: boolean;
 
-    onUpdate: (updated: Item[]) => void;
-    popupWidth?: number | string;
+    onUpdate: (newSettings: TableSetting[]) => void;
+    popupWidth?: DndTreeSelectProps<any>['popupWidth'];
     popupPlacement?: PopperPlacement;
-    getItemTitle?: (item: Item) => TableColumnSetupItem['title'];
-    showStatus?: boolean;
-    className?: string;
 }
 
 export const TableColumnSetup = (props: TableColumnSetupProps) => {
     const {
-        switcher,
         renderSwitcher,
-        disabled,
         popupWidth,
         popupPlacement,
-        className,
         items: propsItems,
-        itemsHeight,
-        getItemTitle = (item: Item) => item.title,
-        sortable = true,
-        filterable = false,
-        showStatus,
+        onUpdate: propsOnUpdate,
+        sortable,
     } = props;
 
-    const [focused, setFocused] = React.useState(false);
-    const [items, setItems] = React.useState<Item[]>([]);
-    const [currentItems, setCurrentItems] = React.useState<Item[]>([]);
-    const [requiredItems, setRequiredItems] = React.useState<Item[]>([]);
+    const [open, setOpen] = React.useState(false);
 
-    const refControl = React.useRef(null);
+    const prepareItems = React.useCallback((tableColumnItems: TableColumnSetupItem[]) => {
+        return tableColumnItems.map<Item>((tableColumnItem) => {
+            const hasSelectionIcon = tableColumnItem.isRequired === false;
 
-    const LIST_ITEM_HEIGHT = 36;
+            return {
+                ...tableColumnItem,
+                startSlot: tableColumnItem.isRequired ? <Icon data={Lock} /> : undefined,
+                isDragDisabled: sortable === false,
+                hasSelectionIcon,
 
-    const getRequiredItems = (list: Item[]) =>
-        list
-            .filter(({required}) => required)
-            .map((column) => ({
-                ...column,
-                disabled: true,
-            }));
+                // to overwrite select bacground effect - https://github.com/gravity-ui/uikit/blob/main/src/components/useList/components/ListItemView/ListItemView.tsx#L125
+                className: hasSelectionIcon ? undefined : requiredDndItemCn,
```

Comment:
better not to pass selected prop in view if you don't need selected behaviour in disabled element

Reply:
I want this item be `selected`, but I do not want default style that impossible to turn off.

I want selected property cause i do not know what side effect this property will trigger in the future and by logic required items always selected. 

Even now [there is](https://github.com/gravity-ui/uikit/blob/88ffdc67918ed5209a640f783df5d4bc00641d7c/src/components/useList/components/ListItemView/ListItemView.tsx#L120) side effect

Reply:
fun type by the way - `bacground`